unified_planning/io/ma_pddl_writer.py

```diff
@@ -0,0 +1,1066 @@
+                                    else:
+                                        out.write(f"(at end {converter.convert(c)})")
+                                else:
+                                    if not interval.is_left_open():
+                                        out.write(f"(at start {converter.convert(c)})")
+                                    out.write(f"(over all {converter.convert(c)})")
+                                    if not interval.is_right_open():
+                                        out.write(f"(at end {converter.convert(c)})")
+                        out.write(")")
+                    if len(a.effects) > 0:
+                        out.write("\n  :effect (and")
+                        for t, el in a.effects.items():
+                            for e in el:
+                                if t.is_from_start():
+                                    out.write(f" (at start")
+                                else:
+                                    out.write(f" (at end")
+                                if e.is_conditional():
+                                    out.write(
+                                        f" (when {converter.convert(e.condition)}"
+                                    )
+                                if e.value.is_true():
+                                    out.write(f" {converter.convert(e.fluent)}")
+                                elif e.value.is_false():
+                                    out.write(f" (not {converter.convert(e.fluent)})")
+                                elif e.is_increase():
+                                    out.write(
+                                        f" (increase {converter.convert(e.fluent)} {converter.convert(e.value)})"
+                                    )
+                                elif e.is_decrease():
+                                    out.write(
+                                        f" (decrease {converter.convert(e.fluent)} {converter.convert(e.value)})"
+                                    )
+                                else:
+                                    out.write(
+                                        f" (assign {converter.convert(e.fluent)} {converter.convert(e.value)})"
+                                    )
+                                if e.is_conditional():
+                                    out.write(f")")
+                                out.write(")")
+                        if a in costs:
+                            out.write(
+                                f" (at end (increase (total-cost) {converter.convert(costs[a])}))"
+                            )
+                        out.write(")")
+                    out.write(")\n")
+                else:
+                    raise NotImplementedError
+            out.write(")\n")
+            out.seek(0)
+            ag_domain = out.read()
+            ag_domains[ag.name] = ag_domain
+            out.truncate(0)
+            out.seek(0)
```

Comment:
See comment below: this is very awkward...

Reply:
Done!